src/sdl2_compat.c

```diff
@@ -6133,11 +6133,22 @@ SDL_DECLSPEC int SDLCALL
 SDL_RenderReadPixels(SDL_Renderer * renderer, const SDL_Rect * rect, Uint32 format, void *pixels, int pitch)
 {
     int result = -1;
+    SDL_Texture* target;
 
     SDL_Surface *surface = SDL3_RenderReadPixels(renderer, rect);
     if (!surface) {
         return -1;
     }
+   
+    if (!format) {
+        target = SDL3_GetRenderTarget(renderer);
+        if (!target) {
+            format = SDL3_GetWindowPixelFormat(SDL3_GetPointerProperty(SDL3_GetRendererProperties(renderer), SDL_PROP_RENDERER_WINDOW_POINTER, NULL));
+        }           
+        else {  
```

Comment:
This patch looks great, and once you match the SDL style, I'll be happy to merge it.

The SDL style is:
```c
if (condition) {
} else {
}
```
Also, usually the if condition is a positive one, so it would be:
```c
if (target) {
} else {
}
```